discord_connector.py

```diff
@@ -309,6 +383,86 @@ def run_in_loop(self, coroutine, action_result, error_message=""):
             return action_result.set_status(phantom.APP_ERROR,
                                             f"Other exception. Error type: {e.__class__.__name__} Details: {str(e)}"), None
 
+    def _handle_on_poll(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        container_count = param.get('container_count', None)
+
+        status, channels = self.run_in_loop(self._guild.fetch_channels(), action_result,
+                                            error_message="Cannot fetch channel from Discord.")
+        if not status:
+            self.save_progress("action result: Cannot Poll messages, unable to fetch channels ")
+            return action_result.set_status(phantom.APP_ERROR, "action result: Cannot Poll messages")
+
+        self._status = self.load_state()
+        last_poll_date = self._state.get("last_poll_date", None)
+        if last_poll_date is not None:
+            last_poll_date_parsing_status, last_poll_date = self.parse_date(last_poll_date)
+            if not last_poll_date_parsing_status:
+                return action_result.set_status(phantom.APP_ERROR, "Unable to parse last poll date")
+
+        self.save_progress("lading last poll date: {}".format(last_poll_date))
+
+        newest_message = last_poll_date
+        if newest_message is None:
+            newest_message = datetime.min.replace(tzinfo=pytz.UTC)
+
+        for channel in filter((lambda channel_to_test: isinstance(channel_to_test, discord.TextChannel)), channels):
+
+            status, messages = self.fetch_message_history(channel.id, action_result, last_poll_date,
+                                                          None, True, container_count)
+            if not status:
+                self.debug_print("action result: Cannot Poll messages from {} {}".format(channel.name, channel.id))
+                return action_result.set_status(phantom.APP_ERROR)
+
+            self.save_progress("saving the containers")
+            for message in messages:
+
+                _, message_creation_date = self.parse_date(message.created_at.strftime("%Y-%m-%d %H:%M:%S"))
```

Comment:
why parse it to string to parse it into datetime to parse it to string again? Can't you simply operate on the `datetime` object already?

Reply:
Parse_date method cuts datetime to seconds instead of miliseconds as this method is used in fetch_message_history and I used it there to remain consistent, I probably should parse datetime to string and string to datetime other way as there is no need for human redable format in on poll action.

Reply:
I would consider leaving it that way, having in mind future refactor because of need to attend more urgent matters.